[Solax X1] RTS support and offline status

SteWers opened the pull request:
## Description:

- Added support for RS485-boards wich need a seperate RTS-line, when sending data
- Added status "off", when there is no response form the inverter

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with Tasmota core ESP32 V.2.0.2
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

SteWers committed:
SolaxX1 RTS #1

SteWers committed:
[Solax X1] RTS support and offline status

SteWers committed:
Merge branch 'arendst:development' into Solax-X1-RTS

SteWers committed:
RTS support and offline status (Fix#1)

SteWers committed:
Merge branch 'Solax-X1-RTS' of https://github.com/SteWers/Tasmota into Solax-X1-RTS

SteWers committed:
Revert my_user_config.h

SteWers committed:
Update my_user_config.h

s-hadinger commented on tasmota/tasmota_template_legacy.h:
```diff
@@ -190,6 +190,7 @@ enum LegacyUserSelectablePins {
   GPI8_OLED_RESET,     // OLED Display Reset
   GPI8_SOLAXX1_TX,     // Solax Inverter tx pin
   GPI8_SOLAXX1_RX,     // Solax Inverter rx pin
+  GPI8_SOLAXX1_RTS,    // Solax Inverter RTS pin
```
Thanks for your PR. You can't just insert new GPIO codes in the existing list, as it would break any existing template. You need to add new GPIOs at the end of the list.

arendst replied:
In addition, no changes to tasmota_template_legacy.h supported. Won't be used as there is no legacy code using it.

Pls remove any changes here.

s-hadinger commented on tasmota/tasmota_template_legacy.h:
```diff
@@ -420,6 +421,7 @@ const uint16_t kGpioConvert[] PROGMEM = {
   AGPIO(GPIO_OLED_RESET),     // OLED Display Reset
   AGPIO(GPIO_SOLAXX1_TX),     // Solax Inverter tx pin
   AGPIO(GPIO_SOLAXX1_RX),     // Solax Inverter rx pin
+  AGPIO(GPIO_SOLAXX1_RTS),    // Solax Inverter RTS pin
```
Same here, please move to the end of the list.

arendst replied:
Same here. Not supported

arendst commented on tasmota/xnrg_12_solaxX1.ino:
```diff
@@ -155,10 +155,18 @@ void solaxX1_RS485Send(uint16_t msgLen)
   solaxX1Serial->write(highByte(crc));
   solaxX1Serial->write(lowByte(crc));
+  solaxX1Serial->flush();
+  if (PinUsed(GPIO_SOLAXX1_RTS)) {
```
Flush doesn't do anything here. It's a receive buffer flush. NOT a send buffer flush (Changed by Core team since a year I must confess)

SteWers committed:
Rework for PR #14305

- removed all changes in tasmota_template_legacy.h
- moved new GPIO at the end of the list in  tasmota_template.h
- removed unnecessary "flush" in xnrg_12_solaxX1.ino

SteWers commented:
Thanks for your comments. I've implemented your suggestions.
